flatfs.go

```diff
@@ -33,6 +58,13 @@ type Datastore struct {
 
 	// sychronize all writes and directory changes for added safety
 	sync bool
+
+	diskUsage int64
+
+	// opMap handles concurrent write operations (put/delete)
+	// to the same key
+	opMap     map[string]*sync.Mutex
```

Comment:
It might be better to use concurrent map. It should be essentially free when we are doing concurrent operations on different keys.

Reply:
:+1: although I think the OS will be the bottleneck here.